File: app/src/main/java/vn/tiki/android/androidhometest/util/StringUtil.java
package vn.tiki.android.androidhometest.util;

public class StringUtil {

    public static String convertPrice(double price, String unit){
        int decimal = (int) price / 1000;
        String mod = String.valueOf((int) (price % 1000));
        if(mod.equals("0")){
            mod = "000";
        }
        return decimal + "." + mod + " " + unit;
    }
}

File: app/src/main/java/vn/tiki/android/androidhometest/util/StringUtil.kt
package vn.tiki.android.androidhometest.util

object StringUtil {

    fun convertPrice(price: Double, unit: String): String {
        val decimal = price.toInt() / 1000
        var mod = (price % 1000).toInt().toString()
        if (mod == "0") {
            mod = "000"
        }
        return decimal.toString() + "." + mod + " " + unit
    }
}
